fury/shaders/base.py

```diff
@@ -200,6 +221,50 @@ def cbk(caller, event, calldata=None):
     return id_observer
 
 
+def shader_apply_effects(
+        showm, actor, effect=None, effects=None, priority=0):
+    """This applies a specific opengl state (effect) or a list of effects just
+    before the actor's shader be executed.
+
+    Parameters
+    ----------
+    showm : fury.window.ShowManager
+    actor : vtk actor
+    effect :  function
+        a function with a glState as argument
+    effects : a list of functions, optional
+    priority : float, optional
+        Related with the shader callback command.
+        Effects with a higher priority are applied first and
+        can be override by the others.
+
+    Returns
+    -------
+    id_observer : int
+        An unsigned Int tag which can be used later to remove the event
+        or retrieve the vtkCommand used in the observer.
+        See more at: https://vtk.org/doc/nightly/html/classvtkObject.html
+
+    """
+    if effects is None:
+        effects = [effect]
+
+    def callback(
+            _caller, _event, calldata=None,
+            effects=[], showm=None):
```

Comment:
you should avoid `[]` as the default parameter.

For your information: https://docs.python-guide.org/writing/gotchas/